design/keycloak.x/configuration.md

```diff
@@ -0,0 +1,159 @@
+# Keycloak.X Server Configuration
+
+* **Status**: Draft #1
+
+## How to set configuration
+
+Keycloak.X can be configured in a property file, using command-line arguments or environment variables. All configuration
+properties are available in all different approaches.
+
+Properties (conf/keycloak.properties):
+
+    hostname.frontendUrl=https://mykeycloak
+    
+Environment variables:
+
+    export KC_HOSTNAME_FRONTEND-URL=https://mykeycloak
```

Comment:
We should probably unify underscores (`_`) and hyphens (`-`). Spring uses a pretty nice convention ([link](https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-external-config-relaxed-binding-from-environment-variables))

Reply:
I think the options are to just convert camelCase to CAMELCASE, or do CAMEL-CASE. I probably perfer CAMELCASE to be honest. Doing CAMEL_CASE is not a good approach as there's then no differentiation between "my.name" and "myName" when converted.

Reply:
I agree with aligning with whatever Quarkus does.

Reply:
Changing to _